app/services/cart_items/create_or_increment.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+require 'dry/monads'
+
+module CartItems
+  class CreateOrIncrement
+    include Dry::Monads[:result, :do]
+
+    def call(params:, user:)
+      product = yield find_product(params[:product_id])
+      quantity = yield get_quantity(params)
+      cart_item = CartItem.find_by(product_id: product.id, user_id: user.id)
+
+      if cart_item
+        yield increase_quantity(cart_item, quantity)
+      else
+        yield create_cart_item(product.id, user.id, quantity)
+      end
+      Success("Added #{product.name} to the cart")
+    end
+
+    private
+
+    def find_product(id)
+      product = Product.find_by(id: id)
+
+      if product
+        Success(product)
+      else
+        Failure('Product does not exist')
+      end
+    end
+
+    def get_quantity(params)
+      quantity = if params[:quantity]
```

Comment:
```suggestion
      quantity =  params[:quantity] ? params[:quantity].to_i : 1
```
IMO 1 shouldn't be hardcoded as integer it will be better to move it to constant

Reply:
Oneliner if will look better imo second one either and in previous method same thing

Reply:
Also instead of 2 if here I'll prefer to move it to method `quantity(quantity_patams)`